internal/core/src/query/visitors/ExecExprVisitor.cpp

```diff
@@ -222,6 +226,114 @@ ExecExprVisitor::ExecUnaryRangeVisitorDispatcher(UnaryRangeExpr& expr_raw) -> Bi
 }
 #pragma clang diagnostic pop
 
+#pragma clang diagnostic push
+#pragma ide diagnostic ignored "Simplify"
+template <typename T>
+auto
+ExecExprVisitor::ExecBinaryArithOpEvalRangeVisitorDispatcher(BinaryArithOpEvalRangeExpr& expr_raw) -> BitsetType {
+    auto& expr = static_cast<BinaryArithOpEvalRangeExprImpl<T>&>(expr_raw);
+    using Index = scalar::ScalarIndex<T>;
+    auto arith_op = expr.arith_op_;
+    auto right_operand = expr.right_operand_;
+    auto op = expr.op_type_;
+    auto val = expr.value_;
+
+    switch (op) {
+        case OpType::Equal: {
```

Comment:
Besides `==` and `!=`, should we support `<`, `<=`, `>`, `>=` also?

Reply:
Yes, I am thinking of adding these additional comparison operators in a separate PR. I will need to think more about how to better integrate these into the overall architecture of the parser. And potentially refactor the current "==" and "!=" handling, if needed. I hope this is fine. :)